File: asyncfactor.py
import asyncio
import aiohttp
import json
import sys
from random import randint

API_URL = "https://0bdwnj5rj7.execute-api.us-east-1.amazonaws.com/prod"


async def call_factor_API(session,num, data):
    print(f"Factoring {num}...")
    async with session.get(url=API_URL, json=data) as resp:
        result = await resp.text()
        all_factors = json.loads(json.loads(result)['body'])['result']

        all_factors = map(str, all_factors)
        print(f"{num} can be factored into : {', '.join(all_factors)}")

async def main():
    #N = int(input("How many numbers to factor?"))
    N = int(sys.argv[1])
    numbers = [randint(5000, 10000) for _ in range(N)]
    tasks = []
    async with aiohttp.ClientSession() as session:
        for n in numbers:
            tasks.append(asyncio.ensure_future(call_factor_API(session,n,{"number":n})))

        await asyncio.gather(*tasks)

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
loop.close()

File: asyncfactor.js
const axios = require("axios");
const API_URL = "https://0bdwnj5rj7.execute-api.us-east-1.amazonaws.com/prod"

async function callFactorAPI(num, data){
  console.log(`Factoring ${num}`);
  var result = await axios({
     method: 'get',
     url: API_URL,
     data: data
  });
  return JSON.parse(result.data.body).result
}

(async function(){
 var N = parseInt(process.argv.slice(2)[0]);
 var numsToFactor = Object.keys([...Array(N)]).map(x => Math.floor(Math.random() * 5000) + 5000);
 try{
 var factoredValues = await Promise.all(numsToFactor.map(num => callFactorAPI(num, {number:num})));
 }catch(e){
  console.log(e);
 }
 Object.keys(numsToFactor).forEach(n => {
   console.log(`${numsToFactor[n]} can be factored into ${factoredValues[n]}`);
 });
})();
